HW05-byteCodes/src/main/java/ru/mts/MyIOC.java

```diff
@@ -0,0 +1,62 @@
+package ru.mts;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyIOC {
+    static ArrayList<Method> loggedMethods;
+
+    public MyIOC() {
+    }
+
+    private static List<Method> getAnnotationMethods(Object object, Class classAnnotation) {
+        ArrayList<Method> varAnnotationMethod = new ArrayList(10);
+        for (Method method : object.getClass().getDeclaredMethods()) {
+            if (method.isAnnotationPresent(classAnnotation)) {
+                varAnnotationMethod.add(method);
+            }
+        }
+        return (List<Method>) varAnnotationMethod;
+    }
+
+    static <T> T createLogging(T object) throws Exception {
+        InvocationHandler handler = new MyInvocationHandler(object);
+        loggedMethods = (ArrayList<Method>) getAnnotationMethods(object, Log.class);
```

Comment:
```suggestion
        loggedMethods = getAnnotationMethods(object, Log.class);
```

Надо `loggedMethods` сделать `List<Method>` и приведения не понадобится.